MapboxNavigation/Style.swift

```diff
@@ -0,0 +1,311 @@
+            Button.appearance(for: traitCollection).textColor = color
+        }
+        
+        if let color = lineColor {
+            // Line view can be a dashed line view so we add a lineColor to avoid changing the background.
+            LineView.appearance(for: traitCollection).lineColor = color
+            // Separator view can also be a divider (vertical/horizontal) with a solid background color.
+            SeparatorView.appearance(for: traitCollection).backgroundColor = color
+        }
+        
+        // Maneuver page view controller
+        
+        if let color = maneuverViewBackgroundColor {
+            ManeuverView.appearance(for: traitCollection).backgroundColor = color
+        }
+        
+        if let height = maneuverViewHeight {
+            ManeuverView.appearance(for: traitCollection).height = height
+        }
+        
+        // Table view (drawer)
+        
+        if let color = headerBackgroundColor {
+            RouteTableViewHeaderView.appearance(for: traitCollection).backgroundColor = color
+        }
+        
+        if let font = cellTitleLabelFont {
+            CellTitleLabel.appearance(for: traitCollection).font = font
+        }
+        
+        if let color = cellTitleLabelTextColor {
+            CellTitleLabel.appearance(for: traitCollection).textColor = color
+        }
+        
+        if let font = cellSubtitleLabelFont {
+            CellSubtitleLabel.appearance(for: traitCollection).font = font
+        }
+        
+        if let color = cellSubtitleLabelTextColor {
+            CellSubtitleLabel.appearance(for: traitCollection).textColor = color
+        }
+    }
+}
+
+@objc(MBButton)
+public class Button: StylableButton { }
+
+@objc(MBStylableLabel)
+public class StylableLabel : UILabel { }
+
+@objc(MBTitleLabel)
+public class TitleLabel: StylableLabel { }
+@objc(MBSubtitleLabel)
+public class SubtitleLabel: StylableLabel { }
+
+@objc(MBCellTitleLabel)
+public class CellTitleLabel: StylableLabel { }
+@objc(MBCellSubtitleLabel)
+public class CellSubtitleLabel: StylableLabel { }
+
+@objc(MBProgressBar)
+public class ProgressBar: UIView { }
+
+@objc(MBLineView)
+public class LineView: UIView {
+    dynamic var lineColor: UIColor = .defaultLine {
+        didSet {
+            setNeedsDisplay()
+            setNeedsLayout()
+        }
+    }
+}
+
+@objc(MBSeparatorView)
+public class SeparatorView: UIView { }
+
+@objc(MBStylableButton)
+public class StylableButton: UIButton {
+    dynamic var textColor: UIColor = .defaultPrimaryText {
+        didSet {
+            setTitleColor(textColor, for: .normal)
+        }
+    }
+}
+
+@objc(MBManeuverView)
+class ManeuverView: UIView {
+    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
+    
+    dynamic var height: CGFloat = .defaultManeuverViewHeight {
+        didSet {
+            heightConstraint.constant = height
+            setNeedsUpdateConstraints()
+        }
+    }
+}
+
+class RouteStyleLayer: MGLLineStyleLayer {
```

Comment:
I overlooked these delegate methods. Not sure if we should keep them or make them stylable by UIAppearance. The problem is that only UIViews have proper support for UIAppearance. You can achieve some of it by using Objective-C's `UI_APPEARANCE_SELECTOR` (or just dynamic in Swift) but it's just a lucky side effect from MGLMapView that makes it update using the appearance when changing size classes.

TBH, I think we should remove this from the style and keep the delegate methods.